ggplot/ggplot.py

```diff
@@ -122,417 +84,389 @@ class _empty(object):
-                        for lab in plt.yticks()[1]:
-                            lab = lab.get_text()
-                            lab = self.ytick_labels.get(lab)
-                            labs.append(lab)
-                        ax.yaxis.set_ticklabels(labs)
-                    elif isinstance(self.ytick_labels, list):
-                        ax.yaxis.set_ticklabels(self.ytick_labels)
-                if self.ytick_formatter:
-                    ax.yaxis.set_major_formatter(self.ytick_formatter)
-                if self.xlimits:
-                    limits = [x if x != None else x_def for x, x_def in zip(self.xlimits, ax.get_xlim())]
-                    if not self.xbreaks and not self.xtick_labels:
-                        labs, minval, maxval= calc_axis_breaks_and_limits(limits[0], limits[1])
-                        ax.xaxis.set_ticks(labs)
-                        ax.xaxis.set_ticklabels(labs)
-                    ax.set_xlim(limits)
-                if self.ylimits:
-                    limits = [y if y != None else y_def for y, y_def in zip(self.ylimits, ax.get_ylim())]
-                    if not self.ytick_labels:
-                        labs, minval, maxval= calc_axis_breaks_and_limits(limits[0], limits[1])
-                        ax.yaxis.set_ticks(labs)
-                        ax.yaxis.set_ticklabels(labs)
-                    ax.set_ylim(limits)
-                if self.scale_y_reverse:
-                    ax.invert_yaxis()
-                if self.scale_x_reverse:
-                    ax.invert_xaxis()
-                if self.scale_y_log:
-                    ax.set_yscale('log', basey=self.scale_y_log)
-                if self.scale_x_log:
-                    ax.set_xscale('log', basex=self.scale_x_log)
-                if self.coord_equal:
-                    min_val = np.min([np.min(ax.get_yticks()), np.min(ax.get_xticks())])
-                    max_val = np.max([np.max(ax.get_yticks()), np.max(ax.get_xticks())])
-                    ax.set_xticks(np.linspace(min_val, max_val, 7))
-                    ax.set_yticks(np.linspace(min_val, max_val, 7))
-
-            # TODO: Having some issues here with things that shouldn't have a
-            # legend or at least shouldn't get shrunk to accomodate one. Need
-            # some sort of test in place to prevent this OR prevent legend
-            # getting set to True.
-            if self.legend:
-                # works with faceted and non-faceted plots
-                ax = axs[0][self.n_rows - 1]
-                box = ax.get_position()
-                ax.set_position([box.x0, box.y0, box.width * 0.8, box.height])
-                add_legend(self.legend, ax)
-
-            # Finaly apply any post plot callbacks (theming, etc)
-            for ax in plt.gcf().axes:
-                self.theme.post_plot_callback(ax)
+    def render(self):
+        """
+        Render the complete plot and return the matplotlib figure
+        """
+        plt.close("all")
```

Comment:
IMO `plt.close("all")` should not be used at all by library code: if do some matplotlib first and then use ggplot, my manual figures are all gone...

My opinion is also that all 'plt' usage should be gone and figures/axis (.> The OO interface of matplotlib) should be used directly. If one is to use gridspec, this makes it possible to pass in a gridspec slice instead of a figure and get "pasting together" of two ggplots in one figure for free :-)

See here for the idea: https://github.com/mwaskom/seaborn/issues/399#issuecomment-84035872